### ISSUE TITLE
OAuth2: persist cookies during authorization_code

### ISSUE THREAD

**Issue by JoniJnm** [usebruno/bruno#3584]
### I have checked the following:

- [X] I've searched existing issues and found nothing related to my issue.

### Describe the feature you want to add

I would like to persist cookies when I'm using OAuth2 authorization_code. So if I close bruno-app and open it again I don't need to login again in the authorizer server.

I image is harder but I could be ideal if instead of opening an in-app browser it opens an external one, where the login is already done.

### Mockups or Images of the feature

No needed? Maybe a boolean option in Preferences to choose "persist cookies/session".

**Comment by Its-treason** [comment]
Currently the Browser `partition` (Where cookies are saved) is hash that is connected to the collection id, so its always the same.

I think the only thing missing is prefixing it with `persist:` so the data is saved on disk. Here: https://github.com/usebruno/bruno/blob/fd22ff89625c88150f4debf1d0b2ff5c200940e0/packages/bruno-electron/src/store/oauth2.js#L53

**Comment by JoniJnm** [comment]
I'm not sure if that is enough https://github.com/electron/electron/issues/9995

**Comment by Its-treason** [comment]
> I'm not sure if that is enough [electron/electron#9995](https://github.com/electron/electron/issues/9995)

Good catch, session cookies are only stored for a session, like the browser does. We would have to save them to disk manually. But i don't think this should be the default.

Do you often have to restart Bruno?

**Comment by JoniJnm** [comment]
I use bruno, and then close it.

**Comment by pooja-bruno** [comment]
Closing this as it has been added in v2.10.0
